api/views/search.py

```diff
@@ -1,30 +1,33 @@
 from api import app, db
-from api.models import Story, StoryPOI, POI
-from api.utils import create_response, row_constructor
+from api.models import POI
+from api.utils import create_response
 from flask import Blueprint, request
 
-SEARCH_URL = "/search"
+SEARCH_POIS_URL = "/search/pois"
 
 mod = Blueprint('search', __name__)
 
-@app.route(SEARCH_URL, methods=['GET'])
+@app.route(SEARCH_POIS_URL, methods=['GET'])
 def search_pois():
     data = request.args
-    pois = []
-    pois1 = []
-    pois2 = []
+    pois_by_name = []
+    pois_by_description = []
 
     if data.get('q') is not None:
         if data.get('name') is not None:
-            if data.get('name') == "True":
-                pois1 = POI.query.filter(POI.name.contains(data['q']))
+            if data.get('name') == "true":
+                pois_by_name = POI.query.filter(POI.name.contains(data['q']))
+        else:
+            return create_response(status=404, message='Missing name parameter (true/false)')
         if data.get('description') is not None:
-            if data.get('description') == "True":
-                pois2 = POI.query.filter(POI.description.contains(data['q']))
+            if data.get('description') == "true":
+                pois_by_description = POI.query.filter(POI.description.contains(data['q']))
+        else:
+            return create_response(status=404, message='Missing description parameter (true/false)')
+    else:
+        return create_response(status=404, message='Missing query parameter')
```

Comment:
i think you can reduce the amount of nested `if`s if you switch the order in which you process things. if you handle the "not" cases first, you don't need an `else` clause since you return something in each condition branch. 

for example:

```python
if something is not None:
    do(something)
else:
    return create_response(...)
```

is the same as:

```python
if something is None:
    return create_response(...)
do(something)
```

Reply:
doing that will also allow you to limit having to do both `if data.get('name') is not None` and `if data.get('name') == "true"` checks. since you would've already handled the `if data.get('name') is None` case, you can take advantage of how the `.get()` function returns `None` if the key doesn't exist _or_ the value doesn't exist. that should allow you to remove another level of nested `if/else`s